### PR TITLE
refactor(dms/rocketmq_instance): fix dms rocketmq instance lint issues

### PR DESCRIPTION
**Pull request by houpeng80**
**What this PR does / why we need it**:
   fix dms rocketmq instance lint issues
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
   fix dms rocketmq instance lint issues
```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/dms/' TESTARGS='-run TestAccDmsRocketMQInstance_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dms/ -v -run TestAccDmsRocketMQInstance_basic -timeout 360m -parallel 4
=== RUN   TestAccDmsRocketMQInstance_basic
=== PAUSE TestAccDmsRocketMQInstance_basic
=== CONT  TestAccDmsRocketMQInstance_basic
--- PASS: TestAccDmsRocketMQInstance_basic (720.13s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dms       720.180s
```

### CODE REVIEW

File: huaweicloud/services/dms/resource_huaweicloud_dms_rocketmq_instance.go [hunk 1]
```diff
@@ -416,7 +417,7 @@ func resourceDmsRocketMQInstanceUpdate(ctx context.Context, d *schema.ResourceDa
 	return resourceDmsRocketMQInstanceRead(ctx, d, meta)
 }
 
-func buildUpdateRocketmqInstanceBodyParams(d *schema.ResourceData, config *config.Config) map[string]interface{} {
+func buildUpdateRocketmqInstanceBodyParams(d *schema.ResourceData, _ *config.Config) map[string]interface{} {
```
Comment (Lance52259) [inline huaweicloud/services/dms/resource_huaweicloud_dms_rocketmq_instance.go]: If the config is not used, we should remove it from the argument list.

**Comment by chengxiangdong** [comment]
A confusing title that should be `refactor(dms/rocketmq_instance)...`

File: huaweicloud/services/dms/resource_huaweicloud_dms_rocketmq_instance.go [hunk 2]
```diff
@@ -580,7 +582,7 @@ func resourceDmsRocketMQInstanceDelete(ctx context.Context, d *schema.ResourceDa
 	return nil
 }
 
-func DmsRocketmqInstanceStateRefreshFunc(client *golangsdk.ServiceClient, instanceID string) resource.StateRefreshFunc {
+func RocketmqInstanceStateRefreshFunc(client *golangsdk.ServiceClient, instanceID string) resource.StateRefreshFunc {
```
Comment (chengxiangdong) [inline huaweicloud/services/dms/resource_huaweicloud_dms_rocketmq_instance.go]:
```suggestion
func rocketMQInstanceStateRefreshFunc(client *golangsdk.ServiceClient, instanceID string) resource.StateRefreshFunc {
```
It seems it should be a private function

**Review by chengxiangdong — COMMENTED** [review]
I think it is better to define `"dmsv2"` as a constant, although it is an automatically generated code.

File: huaweicloud/services/dms/resource_huaweicloud_dms_rocketmq_instance.go (hunk 1 above)
Comment (houpeng80) [reply huaweicloud/services/dms/resource_huaweicloud_dms_rocketmq_instance.go]: done

File: huaweicloud/services/dms/resource_huaweicloud_dms_rocketmq_instance.go (hunk 2 above)
Comment (houpeng80) [reply huaweicloud/services/dms/resource_huaweicloud_dms_rocketmq_instance.go]: done

**Comment by chengxiangdong** [comment]
pls rebase your code

File: docs/resources/dms_rocketmq_instance.md [hunk 3]
```diff
@@ -153,6 +145,7 @@ In addition to all arguments above, the following attributes are exported:
 
 The `cross_vpc_accesses` block supports:
 
+* `advertised_ip` -(Optional, String) The advertised IP Address or domain name.
```
Comment (ShiChangkuo) [inline docs/resources/dms_rocketmq_instance.md]:
```suggestion
* `advertised_ip` - The advertised IP Address or domain name.
```

**Pull request merged**